enterprise/server/backends/userdb/userdb.go

```diff
@@ -931,12 +915,60 @@ func (d *UserDB) GetImpersonatedUser(ctx context.Context) (*tables.User, error)
 	return user, err
 }
 
-func (d *UserDB) DeleteUser(ctx context.Context, userID string) error {
-	// Permission check.
-	_, err := d.GetUserByID(ctx, userID)
+type userGroupRole struct {
+	UserUserID string
+	tables.GroupRole
+}
+
+func (d *UserDB) GetUsersBySubIDPrefixWithoutAuthCheck(ctx context.Context, subIDPrefix string) ([]*tables.User, error) {
+	rq := d.h.NewQuery(ctx, "userdb_get_users_by_sub_id_prefix").Raw(`
+			SELECT * FROM "Users" WHERE sub_id LIKE ?
+	`, subIDPrefix+"%")
+	users, err := db.ScanAll(rq, &tables.User{})
 	if err != nil {
-		return err
+		return nil, err
 	}
+
+	byID := map[string]*tables.User{}
+	for _, u := range users {
+		byID[u.UserID] = u
+	}
```

Comment:
nit: the map makes the ordering non-deterministic which might make this harder to debug (we've been saved once before by a certain query being deterministic). One option is changing the loop to

```
for users := users; len(users) > 0; {
```